### PR TITLE
Add support for handling paginated odata responses

### PR DESCRIPTION
**Pull request by anna-follestad-4ss**
## Description
- add function get_page to authenticate.py
- add unit tests to test_authenticator.py

## Checklist
- [ ] PR title is descriptive and fit for injection into release notes (see tips below)
- [ ] Correct label(s) are used

PR title tips:
* Use imperative mood
* Describe the motivation for change, issue that has been solved or what has been improved - not how
* Examples:
  * Add functionality for Allan variance to sensor_4s.simulate
  * Upgrade to support Python 9.10
  * Remove MacOS from CI

### CODE REVIEW

File: fourinsight/api/authenticate.py [hunk 1]
```diff
@@ -210,6 +210,36 @@ def _update_args_kwargs(self, args, kwargs):
...
+
+        while url:
+            response = self.get(url, **kwargs)
+
+            try:
+                url = response.json().get("@odata.nextLink")
+            except AttributeError:
+                url = None
+            finally:
+                yield response
```
Comment (ace-e4s) [inline fourinsight/api/authenticate.py:241]:
```suggestion
        while url:
            response = self.get(url, **kwargs)

            try:
                response.raise_for_status()
            except HTTPError: 
                url = None
            else:
                url = response.json().get("@odata.nextLink")
            yield response
```

Comment (ace-e4s) [reply fourinsight/api/authenticate.py:241]: @vegard-solum-4ss, ref discussion.

**Comment by vegard-solum-4ss** [comment]
Add a section in the documentation about how to get paginated data. Then, this PR is ready to merge

File: docs/user_guide/api_calls.rst [hunk 2]
```diff
@@ -1,18 +1,43 @@
...
 
     # or with relative url
 
-    response = session.get("/v1.0/Campaigns")
+    response = session.get("/v1.1/Campaigns")
 
 
-.. _4Insight REST API: https://4insight.io/#/developer
+Some API endpoints support OData and have paging, which returns a default number of 50 responses per page. 
+For these endpoints, the ``get_pages`` method can be useful:
```
Comment (ace-e4s) [inline docs/user_guide/api_calls.rst:18]:
```suggestion
Some API endpoints support OData and have paging, which returns a limited number of responses per page. 
For these endpoints, the ``get_pages`` method can be useful:
```

**Review by ace-e4s — APPROVED** [review]

**Pull request merged**